### PR TITLE
Ethereum json rpc

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/ethereum_json_rpc/src/main.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from kybra import Async, CanisterResult, ic, init, nat32, query, update
+from kybra.canisters.management import management_canister
+from kybra.canisters.management.http import HttpResponse, HttpTransformArgs
+from typing import TypeAlias, TypedDict
+
+JSON: TypeAlias = str
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:6]: Why the explicit `TypeAlias` annotation?

File: examples/ethereum_json_rpc/src/main.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+@init
+def init_(ethereum_url: str):
+    stable_storage['ethereum_url'] = ethereum_url
+
+
+@update
+def eth_get_balance(ethereum_address: str) -> Async[JSON]:
+    max_response_bytes = 1_000
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:23]: In Azle this is set to 200?

File: examples/ethereum_json_rpc/src/main.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    cycle_cost_base = 400_000_000
+    cycle_cost_per_byte = 300_000  # TODO not sure on this exact cost
+    cycle_cost_total = cycle_cost_base + cycle_cost_per_byte * max_response_bytes
+
+    http_result: CanisterResult[HttpResponse] = yield management_canister.http_request({
+        'url': stable_storage['ethereum_url'],
+        'max_response_bytes': max_response_bytes,
+        'method': {'post': None},
+        'headers': [],
+        'body': '{{"jsonrpc":"2.0","method":"eth_getBalance","params":["{address}","earliest"],"id":1}}'.format(address=ethereum_address).encode('utf-8'),
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:35]: f-string

File: examples/ethereum_json_rpc/src/main.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        'transform': {
+            'function': ((ic.id()), 'eth_transform'),
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:37]: Why the extra set of parentheses around `ic.id()`?

File: examples/ethereum_json_rpc/src/main.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            'context': bytes()
+        },
+    }).with_cycles(cycle_cost_total)
+
+    if http_result.err is not None:
+        if http_result.err:
+            ic.trap(http_result.err)
+        ic.trap('http_result had an error')
+
+    return http_result.ok['body'].decode('utf-8', 'strict')
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:47]: What is the strict setting for?

File: examples/ethereum_json_rpc/src/main.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    cycle_cost_base = 400_000_000
+    cycle_cost_per_byte = 300_000  # TODO not sure on this exact cost
+    cycle_cost_total = cycle_cost_base + cycle_cost_per_byte * max_response_bytes
+
+    http_result: CanisterResult[HttpResponse] = yield management_canister.http_request({
+        'url': stable_storage['ethereum_url'],
+        'max_response_bytes': max_response_bytes,
+        'method': {'post': None},
+        'headers': [],
+        'body': '{{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["{block_number}", false],"id":1}}'.format(block_number=hex(number)).encode('utf-8'),
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:64]: f-string

File: examples/ethereum_json_rpc/src/main.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        'transform': {
+            'function': ((ic.id()), 'eth_transform'),
```
Comment (lastmjs) [inline examples/ethereum_json_rpc/src/main.py:66]: Extra set of parentheses around `ic.id()`?

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/ethereum_json_rpc/src/main.py (hunk 1 above)
Comment (bdemann) [reply examples/ethereum_json_rpc/src/main.py:6]: I just like being explicit, I can get rid of that

File: examples/ethereum_json_rpc/src/main.py (hunk 5 above)
Comment (bdemann) [reply examples/ethereum_json_rpc/src/main.py:47]: its the error handling scheme. strict is the default so i could just leave it off if you prefer

**Pull request merged**